connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkResult.scala

```diff
@@ -60,13 +61,19 @@ private[sql] class SparkResult[T](
   private def processResponses(stopOnFirstNonEmptyResponse: Boolean): Boolean = {
     while (responses.hasNext) {
       val response = responses.next()
+      if (response.hasSchema) {
+        structType =
```

Comment:
What is the difference between this schema and the one in the arrow batch?

Reply:
It is the original schema and the one in the arrow batch is modified to deduplicate the struct field names.
Also the original schema contains UDT if it's supported. Python client works fine with that.

Reply:
This logic actual becomes more confusing now about the structType assingment.

I am wondering if it should becomes something like
```
if (response.hasSchema)
else if (response.hasArrowBatch)
```

Reply:
Now that the correct schema arrives earlier than arrow batches, we should use it if it's available; otherwise fallback to the schema from arrow batch.

Yes, the response schema and arrow schema could be inconsistent in terms of the nested field names if there are duplicates, but it's not problem while encoder is handling the `ColumnarBatch` as long as the data structure is consistent.